### PR TITLE
Question service: Display interview question on Room page

### PR DESCRIPTION
**Pull request by hsiaojietng**
- 1) Create Question model and controller to allow the creation, deletion and obtaining of interview questions stored in MongoDB.

- 2) Populate Question schema with Questions through https://htmlpreview.github.io/?https://github.com/Bishalsarang/Leetcode-Questions-Scraper/blob/master/assets/sample%20output%20files/out.html and the list of problems in https://leetcode.com/problemset/all/.

- 3) Display Question information on Room page.

**Final result:**
image

**Steps to create Question data**
1. Go on either of the URL
2. Go to the inspect section
3. Copy the HTML for Question content
image
4. In Postman, set this up based on the details of the question
image
5. Copy this section of code into the 'Pre-request Script' tab in Postman
`pm.environment.set("qnsContent", JSON.stringify(
<Paste the HTML code you copied in step 3>));`
6. Lastly POST the request to create the Question.

### CODE REVIEW

File: user-service/model/repository.js [hunk 1]
```diff
@@ -34,4 +34,4 @@ export async function updateAccountByUsername(username, newProfile) {
 
 export async function deleteAccountByUsername(username) {
   return await UserModel.deleteOne({ username });
-}
+}
```
Comment (pualixue) [inline user-service/model/repository.js:37]: Might want to add newline at end of file

**Review by pualixue — CHANGES_REQUESTED** [review]
LGTM, good job for including comments for your code! Just some nits and changing of port number.

File: README.md [hunk 2]
```diff
@@ -16,3 +16,11 @@ This is the repository for CS3219 Group 28 PeerPrep project.
...
 1. Install npm packages using `npm i`.
 2. Run Frontend using `npm start`.
+
+## Question Service
+This service is used to Create, Remove, Get the interview questions in the Room page.
+1. Rename `.env.sample` file to `.env`.
+2. Create a Cloud DB URL using Mongo Atlas.
+3. Enter the DB URL created as `DB_CLOUD_URI` in `.env` file.
+4. Install npm packages using `npm i`.
+5. Run User Service using `npm start`.
```
[the remark this answers is not in the document]
Comment (hsiaojietng) [reply README.md:26]: Thanks for this catch!

File: frontend/src/components/ui/question/QuestionBox.js [hunk 3]
```diff
@@ -1,45 +1,100 @@
...
-  });
+  const [title, setTitle] = useState("");
+  const [categories, setCategories] = useState([]);
+  const [content, setContent] = useState("");
+  const [difficulty, setDifficulty] = useState("");
+
+  useEffect(() => {
+    async function fetchData() {
+      await axios
+        .get("http://localhost:8002/api/question/level/Hard") // 'Hard' is hardcoded for now
```
[the remark this answers is not in the document]
Comment (hsiaojietng) [reply frontend/src/components/ui/question/QuestionBox.js:31]: Ah thanks for this, I did not know that!

File: README.md (hunk 2 above)
Comment (pualixue) [inline README.md:26]:
```suggestion
5. Run Question Service using `npm start`.
```
I think you forgot to change this haha

**Review by pualixue — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (hsiaojietng) [reply README.md:26]: Thanks for this haha, I did not save the file.

**Review by pualixue — APPROVED** [review]

**Pull request merged**